lib/cli/run-helpers.js

```diff
@@ -0,0 +1,335 @@
+'use strict';
+
+const fs = require('fs');
+const debug = require('debug')('mocha:cli:run:helpers');
+const Mocha = require('../mocha');
+const path = require('path');
+const utils = require('../utils');
+const minimatch = require('minimatch');
+const chalk = require('chalk');
+const symbols = require('log-symbols');
+
+const cwd = (exports.cwd = process.cwd());
+
+/**
+ * Exits Mocha when tests + code under test has finished execution (default)
+ * @param {number} code - Exit code; typically # of failures
+ * @private
+ */
+const exitLater = code => {
+  process.on('exit', () => {
+    process.exit(Math.min(code, 255));
+  });
+};
+
+/**
+ * Exits Mocha when Mocha itself has finished execution, regardless of
+ * what the tests or code under test is doing.
+ * @param {number} code - Exit code; typically # of failures
+ * @private
+ */
+const exit = code => {
+  const clampedCode = Math.min(code, 255);
```

Comment:
`process.exitCode` should be one of:
* `0` for _success_
* `1` for _failure_ (e.g., internal error, if _any_ tests fail, no tests were run)
* `2` for _failure_ due to invalid command line usage
* `126` if "bin/_mocha" not executable
* `128+n` where `n` is signal# (e.g., `SIGINT`=2, so rc=`130`)

This should be _explicitly_ added to the documentation.

Reply:
if you want to argue for that, please create an issue, but this would break certain CI systems which rely on exit code abuse.

Reply:
Done. #3559